src/Uno.UI/UI/Xaml/DependencyObjectStore.Binder.cs

```diff
@@ -374,8 +374,12 @@ public void SetResourceBinding(DependencyProperty dependencyProperty, object res
 			var precedence = _overriddenPrecedences?.Count > 0
 				? _overriddenPrecedences.Peek()
 				: default;
+			SetResourceBinding(dependencyProperty, resourceKey, isTheme, context, precedence, setterBindingPath: null);
+		}
 
-			var binding = new ResourceBinding(resourceKey, isTheme, context, precedence ?? DependencyPropertyValuePrecedences.Local);
+		internal void SetResourceBinding(DependencyProperty dependencyProperty, object resourceKey, bool isTheme, object context, DependencyPropertyValuePrecedences? precedence, BindingPath? setterBindingPath)
```

Comment:
can `resourceKey` be a string or `SpecializedResourceDictionary.ResourceKey` instead to improve resolution performance ?

Reply:
Probably - I'll take a look

Reply:
moar performance ;)